branchLayoutImpl/src/main/java/com/virtuslab/branchlayout/impl/manager/BranchLayoutFileReader.java

```diff
@@ -1,54 +1,50 @@
-package com.virtuslab.branchlayout.impl;
+package com.virtuslab.branchlayout.impl.manager;
 
-import java.nio.file.Files;
 import java.nio.file.Path;
 
 import io.vavr.Tuple2;
 import io.vavr.collection.Array;
 import io.vavr.collection.List;
-import io.vavr.control.Try;
 import lombok.Getter;
 import lombok.RequiredArgsConstructor;
 import org.checkerframework.checker.index.qual.GTENegativeOne;
 import org.checkerframework.checker.index.qual.NonNegative;
+import org.checkerframework.checker.index.qual.Positive;
 
 import com.virtuslab.branchlayout.api.BaseBranchLayoutEntry;
 import com.virtuslab.branchlayout.api.BranchLayoutException;
-import com.virtuslab.branchlayout.api.IBranchLayoutParser;
+import com.virtuslab.branchlayout.api.manager.IBranchLayoutReader;
+import com.virtuslab.branchlayout.impl.BranchLayout;
+import com.virtuslab.branchlayout.impl.BranchLayoutEntry;
 import com.virtuslab.logger.IPrefixedLambdaLogger;
 import com.virtuslab.logger.PrefixedLambdaLoggerFactory;
 
 @RequiredArgsConstructor
 @Getter
-public class BranchLayoutFileParser implements IBranchLayoutParser {
+public class BranchLayoutFileReader implements IBranchLayoutReader {
   private static final IPrefixedLambdaLogger LOG = PrefixedLambdaLoggerFactory.getLogger("branchLayout");
 
   private final Path path;
-  private Character indentCharacter = ' ';
-  @NonNegative
-  private int levelWidth = 0;
 
-  public BranchLayout parse() throws BranchLayoutException {
-    LOG.debug("Entering");
-    LOG.debug(() -> "Branch layout file path: ${path}");
+  private final char indentCharacter;
+  @Positive
+  private final int indentWidth;
 
-    List<BaseBranchLayoutEntry> roots = List.empty();
-    List<String> lines = getFileLines();
-    List<String> linesWithoutBlank = lines.reject(String::isBlank);
-
-    LOG.debug(() -> "${lines.length()} line(s) fund");
+  public BranchLayout read() throws BranchLayoutException {
+    LOG.debug("Entering");
+    LOG.debug(() -> "Reading branch layout from ${path} with indent character ASCII code = ${(int)indentCharacter} " +
+        "and indent width = ${indentWidth}:");
 
-    deriveIndentCharacter(lines);
+    List<String> lines = BranchLayoutFileUtils.getFileLines(path).reject(String::isBlank);
```

Comment:
You can't do that coz this cause that blank lines are not counted, so error line in exception (and message) is wrong

Reply:
Would be nice to add unit test that would capture that if none exists so far

Reply:
you are right